moab_state_transform.ink

```diff
@@ -0,0 +1,137 @@
+###
+
+# MSFT Bonsai 
+# Copyright 2020 Microsoft
+# This code is licensed under MIT license (see LICENSE for details)
+
+# Moab Sample Using State Transform during Training Only
+
+###
+
+inkling "2.0"
+
+using Math
+using Goal
+
+# Distances measured in meters
+const RadiusOfPlate = 0.1125 # m
+
+# Velocities measured in meters per sec.
+const MaxVelocity = 6.0
+const MaxInitialVelocity = 1.0
+
+# Threshold for ball placement
+const CloseEnough = 0.02
+
+# Default time delta between simulation steps (s)
+const DefaultTimeDelta = 0.045
+
+# Maximum distance per step in meters
+const MaxDistancePerStep = DefaultTimeDelta * MaxVelocity
+
+type SimState {
+    # Ball X,Y position
+    ball_x: number<-MaxDistancePerStep - RadiusOfPlate .. RadiusOfPlate + MaxDistancePerStep>,
+    ball_y: number<-MaxDistancePerStep - RadiusOfPlate .. RadiusOfPlate + MaxDistancePerStep>,
+
+    # Ball X,Y velocity
+    ball_vel_x: number<-MaxVelocity .. MaxVelocity>,
+    ball_vel_y: number<-MaxVelocity .. MaxVelocity>,
+}
+
+# State received from the simulator after each iteration
+type ObservableState {
```

Comment:
This is the same type as SimState — no need to duplicate it.

Reply:
I'm thinking about including target_x and target_y so users can change SimConfig to be non-zero as well

```
# State received from the simulator after each iteration
type SimState {
    # Ball X,Y position
    ball_x: number<-MaxDistancePerStep - RadiusOfPlate .. RadiusOfPlate + MaxDistancePerStep>,
    ball_y: number<-MaxDistancePerStep - RadiusOfPlate .. RadiusOfPlate + MaxDistancePerStep>,

    # Ball X,Y velocity
    ball_vel_x: number<-MaxVelocity .. MaxVelocity>,
    ball_vel_y: number<-MaxVelocity .. MaxVelocity>,

    # Target Stationary X,Y position
    target_x: number<-MaxDistancePerStep - RadiusOfPlate .. RadiusOfPlate + MaxDistancePerStep>,
    target_y: number<-MaxDistancePerStep - RadiusOfPlate .. RadiusOfPlate + MaxDistancePerStep>,
}
```

Reply:
> Made a few suggestions for comment improvements. I also suggest making a sub-folder for example Inkling files, so we can keep adding more without cluttering the root folder.

@shnayder Thanks for the great comments. This folder idea could be great so we can have the freedom to add many more complex inkling samples with moab!